In [1]:
from math import factorial as fl

In [2]:
import numpy as np

### 1) Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.  

$P=P_{100}(85)=C_{85}^{100}⋅0.8^{85}⋅0.2^{15}.$

$C_{85}^{100}= \frac{100!}{85! * (100-85!)}$ 

$C_{85}^{100}= \frac{100!}{85! * 15!}$ 

In [3]:
(fl(100) / (fl(85) * fl(15))) * (0.8**85) * (0.2**15)

#### Ответ: 0.04806

### 2) Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? Какова вероятность, что перегорят ровно две?  

In [4]:
lambd_ = 5000*0.0004

In [5]:
lambd_

2.0

P(m) = ((lambd_**2) / fl(m)) * np.exp(-lambd_)

#### a) какова вероятность, что ни одна из них не перегорит в первый день?

In [6]:
m = 2

In [7]:
((lambd_**m) / fl(m)) * np.exp(-lambd_)

0.2706705664732254

#### Ответ: 0.2706

#### б) какова вероятность, что перегорят ровно две?  

In [8]:
m = 0

In [9]:
((lambd_**m) / fl(m)) * np.exp(-lambd_)

0.1353352832366127

#### Ответ: 0.1353

### 3) Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?  

In [10]:
(fl(144) / (fl(70) * fl(74))) * (0.5**70) * (0.5**74)

0.06281178035144776

#### Ответ: 0.06281

### 4) В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. Какова вероятность того, что все мячи белые? Какова вероятность того, что ровно два мяча белые? Какова вероятность того, что хотя бы один мяч белый?  

для удобства функцию комбинаторики

In [11]:
def C(n, m):
    return fl(m) / (fl(n) * fl(m-n))

#### а) какова вероятность того, что все мячи белые?

$\frac{C^{2}_{7}C^{2}_{9}}{C^{2}_{10}C^{2}_{11}}$

In [12]:
(C(2, 7) * C(2, 9)) / (C(2, 10) * C(2, 11))

0.3054545454545455

#### Ответ: 0.3054

#### б) какова вероятность того, что ровно два мяча белые?

$\frac{C^{2}_{16}}{C^{2}_{10}C^{2}_{11}}$

In [13]:
(C(2, 16)) / (C(2, 10) * C(2, 11))

0.048484848484848485

#### Ответ: 0.04848

#### в) какова вероятность того, что хотя бы один мяч белый?  
От обратного какая вероятность получить 0 черных шаров

$\frac{C^{2}_{3}C^{2}_{2}}{C^{2}_{10}C^{2}_{11}}$

In [14]:
1 - (C(2, 3) * C(2, 2)) / (C(2, 10) * C(2, 11))

0.9987878787878788

#### Ответ: 0.99878